public/scripts/extensions/tts/kokoro.js

```diff
@@ -0,0 +1,203 @@
+import { saveTtsProviderSettings } from './index.js';
+import { KokoroTTS } from '../../../lib/kokoro.web.js';
+
+export class KokoroTtsProvider {
+    constructor() {
+        this.settings = {
+            modelId: "onnx-community/Kokoro-82M-v1.0-ONNX",
+            dtype: "q8",
+            device: "wasm",
+            voiceMap: {},
+            defaultVoice: "af_heart",
+            speakingRate: 1.0,
+            volumeGainDb: 0.0
+        };
+        this.ready = false;
+        this.voices = [
+            "af_heart", "af_alloy", "af_aoede", "af_bella", "af_jessica", "af_kore", "af_nicole", "af_nova", "af_river", "af_sarah", "af_sky", "am_adam", "am_echo", "am_eric", "am_fenrir", "am_liam", "am_michael", "am_onyx", "am_puck", "am_santa", "bf_emma", "bf_isabella", "bm_george", "bm_lewis", "bf_alice", "bf_lily", "bm_daniel", "bm_fable"
+        ];
+        this.tts = null;
+        this.separator = ' ... ... ... ';
+    }
+
+    async loadSettings(settings) {
+        if (settings.modelId !== undefined) this.settings.modelId = settings.modelId;
+        if (settings.dtype !== undefined) this.settings.dtype = settings.dtype;
+        if (settings.device !== undefined) this.settings.device = settings.device;
+        if (settings.voiceMap !== undefined) this.settings.voiceMap = settings.voiceMap;
+        if (settings.defaultVoice !== undefined) this.settings.defaultVoice = settings.defaultVoice;
+    }
+
+    async checkReady() {
+        try {
+            if (!this.tts) {
+                console.log('Initializing Kokoro TTS with settings:', {
+                    modelId: this.settings.modelId,
+                    dtype: this.settings.dtype,
+                    device: this.settings.device
+                });
+
+                // Use KokoroTTS class
+                const tts = await KokoroTTS.from_pretrained(this.settings.modelId, {
+                    dtype: this.settings.dtype,
+                    device: this.settings.device
+                });
+
+                // Check if generate method exists
+                if (typeof tts.generate !== 'function') {
+                    throw new Error('TTS instance does not have generate method');
+                }
+
+                this.tts = tts;
+                console.log('TTS initialized:', this.tts);
+                console.log('Available methods:', Object.getOwnPropertyNames(Object.getPrototypeOf(this.tts)));
+            }
+            this.ready = true;
+            return true;
+        } catch (error) {
+            console.error("Kokoro TTS initialization failed:", error);
+            this.ready = false;
+            return false;
+        }
+    }
+
+    async onRefreshClick() {
+        return await this.checkReady();
+    }
+
+    get settingsHtml() {
+        return `
+            <div class="kokoro_tts_settings">
```

Comment:
Big HTML blocks should go into template files and loaded via `renderExtensionTemplateAsync`.
Supports variables and all, via Handlebar.